src/TextDateOfBirthInput/TextDateOfBirthInput.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import { TextInput } from '../TextInput'
+import { Box } from '../Box'
+import { Fieldset, FieldsetProps } from '../fields/Fieldset'
+import { dateOfBirthValidator } from '../utils/dateOfBirth/dateOfBirthValidator'
+import { createErrorMessage } from '../utils/dateOfBirth/schema'
+
+export interface DateObjectType {
+  day: string
+  month: string
+  year: string
+}
+
+export type DateObject = {
+  [K in keyof DateObjectType]?: DateObjectType[K] | null
+}
+
+export type TextDateOfBirthInputProps = {
+  value: DateObjectType
+  onChange: (value: DateObjectType) => void
+  showCompleted?: boolean
+} & Pick<FieldsetProps, 'label' | 'error' | 'assistiveText'>
+
+/**
+ * Renders a set of input fields to collect a date of birth as separate day, month,
+ * and year fields.
+ *
+ * @function dateOfBirthValidator validates a date of birth provided in a DateObject format:
+ * - All fields are present and correctly formatted.
+ * - The age is at least 17 years old.
+ * - The year is not before 1900.
+ */
+export const TextDateOfBirthInput = ({
```

Comment:
I can see that this is no longer `forwardRef`, I think we need that in order to be able to reuse this in sign-up. 

Treating is as a ref also allows for react-hook-form to then focus the field in case of a validation error.